### PR TITLE
fix: hint-text wrapping and alignment

### PR DESCRIPTION
**Pull request by anuraghazra**
## Description

Fixes: https://razorpay.atlassian.net/browse/DSSUP-69

## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

### CODE REVIEW

File: packages/blade/src/components/Form/FormHint.tsx [hunk 1]
```diff
@@ -23,12 +25,17 @@ const HintText = ({ icon: Icon, children, id, color, size }: HintTextProps): Rea
   return (
     <BaseBox marginTop={hintMarginTop[size]} id={id}>
       <FormHintWrapper>
-        {Icon ? <Icon /> : null}
+        {Icon ? (
+          <Box flexShrink={0} marginTop="spacing.1">
+            <Icon />
+          </Box>
```
Comment (saurabhdaware) [inline packages/blade/src/components/Form/FormHint.tsx:31]: what is spacing.1? add comment maybe?

File: packages/blade/src/components/Form/FormHint.tsx [hunk 2]
```diff
@@ -79,16 +86,18 @@ export type FormHintProps = {
 
 const Icons = {
   error: ({ size }: { size: 'small' | 'medium' | 'large' }): ReactElement => (
-    <>
-      <InfoIcon color="feedback.icon.negative.intense" size={hintIconSize[size]} />
-      <BaseBox marginRight="spacing.2" />
-    </>
+    <InfoIcon
+      display={'block' as never}
```
Comment (saurabhdaware) [inline packages/blade/src/components/Form/FormHint.tsx:90]: why never? this code is used in react native also no?

File: packages/blade/src/components/Form/FormHintWrapper.web.tsx [hunk 3]
```diff
@@ -3,15 +3,7 @@ import BaseBox from '~components/Box/BaseBox';
...
-    <BaseBox
-      as="span"
-      display="flex"
-      flexDirection="row"
-      alignItems="center"
-      style={{
-        wordBreak: 'break-all',
-      }}
-    >
+    <BaseBox as="span" display="flex" flexDirection="row" alignItems="flex-start" gap="spacing.2">
```
Comment (saurabhdaware) [inline packages/blade/src/components/Form/FormHintWrapper.web.tsx:6]: gap wasn't there earlier. is this adding some extra spacing?

Comment (anuraghazra) [reply packages/blade/src/components/Form/FormHintWrapper.web.tsx:6]: It was there. https://github.com/razorpay/blade/blob/b9d4e13725382b07f791ad3b8bf0980e9ae64c7b/packages/blade/src/components/Form/FormHint.tsx#L90

We used spacer components before cuz RN didn't support gap prop

File: packages/blade/src/components/Form/FormHint.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Form/FormHint.tsx:90]: I think it'll just be ignored on RN. 

We do the same on other places as well

https://github.com/razorpay/blade/blob/601cedc8ccf07984a2a2a88b92ca3eb8d28b1f84/packages/blade/src/components/Card/Card.tsx#L204

Comment (anuraghazra) [reply packages/blade/src/components/Form/FormHint.tsx:90]: Our RN setup is totally broken as of now, I do not currently have bandwidth to check and confirm if this works or not.

File: packages/blade/src/components/Form/FormHint.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Form/FormHint.tsx:31]: done

File: packages/blade/src/components/Form/FormHint.tsx (hunk 2 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Form/FormHint.tsx:90]: lol makes sense. yolo

File: packages/blade/src/components/Form/FormHint.tsx (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Form/FormHint.tsx:31]: lies

**Review by saurabhdaware — APPROVED** [review]

**Pull request merged**